sqlglot/dialects/snowflake.py

```diff
@@ -1555,10 +1558,16 @@ def timestampsub_sql(self, expression: exp.TimestampSub):
         def jsonextract_sql(self, expression: exp.JSONExtract):
             this = expression.this
 
-            # JSON strings are valid coming from other dialects such as BQ
+            # JSON strings are valid coming from other dialects such as BQ so
+            # for these cases we PARSE_JSON preemptively
+            if not isinstance(this, (exp.ParseJSON, exp.JSONExtract)) and not expression.args.get(
+                "requires_json"
+            ):
+                this = exp.ParseJSON(this=this)
```

Comment:
Snowflake's `GET_PATH` [supports a few different types for the first argument](https://docs.snowflake.com/en/sql-reference/functions/get_path#arguments). Have you verified that in all cases, wrapping with `PARSE_JSON` is at least a no-op?